File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/dto/PresaleCancellationDetailDto.java
package com.ccadmin.app.sale.model.dto;

import com.ccadmin.app.sale.model.entity.PresaleHeadEntity;

import java.math.BigDecimal;

public class PresaleCancellationDetailDto {

    public PresaleHeadEntity Headboard;
    public SaleDetailDto SaleDetail;
    public boolean HasStockReservation;
    public BigDecimal PendingPaymentAmount;

    public PresaleCancellationDetailDto() {
        this.PendingPaymentAmount = BigDecimal.ZERO;
    }
}

File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/dto/PresaleCancellationDetailDto.ts
import { PresaleHeadEntity } from "../entity/PresaleHeadEntity";
import { SaleDetailDto } from "./SaleDetailDto";

export class PresaleCancellationDetailDto {
    public Headboard: PresaleHeadEntity = new PresaleHeadEntity();
    public SaleDetail: SaleDetailDto | null = null;
    public HasStockReservation: boolean = false;
    public PendingPaymentAmount: number = 0;
}
